superset/models.py

```diff
@@ -1019,7 +1019,9 @@ def query(  # sqla
             raise Exception(_(
                 "Datetime column not provided as part table configuration "
                 "and is required by this type of chart"))
-
+        for m in metrics:
+            if m not in metrics_dict:
+                raise Exception(_("Metric '{}' is not valid".format(m)))
```

Comment:
What causes this condition? Is it [only] when a metric has been deleted? If so it would be nice to make that clear in the message. (if there are other conditions that can create this situation the current message is fine by me)